### PR TITLE
Testing market interaction

### PR DESCRIPTION
**Pull request by TlStephen2011**
Interaction between broker and market is working, broker receiving broadcasts from market via router, need to test random disconnects among clients to see if it still works. Market needs to add executed or rejected logic.

### CODE REVIEW

File: fixme-parent/fixme-broker/src/main/java/za/co/wethinkcode/broker/Broker.java [hunk 1]
```diff
@@ -1,35 +1,122 @@
...
     private final String HOST = "127.0.0.1";
-    Socket socket;
-    
-    public Broker() throws UnknownHostException, IOException{
-    	socket = new Socket(HOST, PORT);
-    	System.out.println("Connection to router has been established");
-    	brokerId = processResponse();
-    	System.out.println("Allocated ID: " + brokerId);    	
+    private Socket socket;
+    private Scanner fromRouter;
```
Comment (diVid3) [inline fixme-parent/fixme-broker/src/main/java/za/co/wethinkcode/broker/Broker.java:23]: This was added as processResponse was creating its own new scanner, thus dropping the state of the scanner created in the constructor, which will contain messages if the broker connects first and then the market.

**Review by diVid3 — CHANGES_REQUESTED** [review]
There seems to be a slight issue when closing the router unexpectedly, causing an input loop at broker CLI.

**Comment by TlStephen2011** [comment]
> There seems to be a slight issue when closing the router unexpectedly, causing an input loop at broker CLI.

**Comment by TlStephen2011** [comment]
hotfix added

**Review by TlStephen2011 — COMMENTED** [review]
Will the line ever equal "", besides when connection to router is broken?

**Comment by diVid3** [comment]
> Will the line ever equal "", besides when connection to router is broken?

No not that I know of.

**Review by diVid3 — APPROVED** [review]
Looking good. Approved 👍

**Review by mmbatha — APPROVED** [review]

**Pull request merged**